cli/compose/loader/volume.go

```diff
@@ -70,6 +70,8 @@ func populateFieldFromBuffer(char rune, buffer []rune, volume *types.ServiceVolu
 		switch option {
 		case "ro":
 			volume.ReadOnly = true
+		case "rw":
```

Comment:
I'm not sure about the technical details, but the `rw` flag used to work before #30597.

From my limited understanding your comment:

> It adds a client-side parser for the old volume spec. As far as I know this is the first time we're attempting to fully parse the spec on the client (without being aware of the engine platform).

and what I understood from the diff it seems to indicate that the volume mount options used to be passed through to the Docker Engine. Is my understanding correct?